packages/tangle/tipmanager.go

```diff
@@ -429,24 +422,24 @@ func (t *TipManager) processMessage(messageID MessageID, messageWalker *walker.W
 	})
 }
 
-func (t *TipManager) checkMarker(marker *markers.Marker, messageWalker *walker.Walker[MessageID], markerWalker *walker.Walker[markers.Marker], minSupportedTimestamp time.Time) (timestampValid bool) {
+func (t *TipManager) checkMarker(marker *markers.Marker, previousMessageID MessageID, messageWalker *walker.Walker[MessageID], markerWalker *walker.Walker[markers.Marker], minSupportedTimestamp time.Time) (messageID MessageID, timestampValid bool) {
 	messageID, messageIssuingTime := t.getMarkerMessage(marker)
 
-	// should never enter this condition as other checks before already cover this case, but leaving it just for safety
+	// marker before minSupportedTimestamp
 	if messageIssuingTime.Before(minSupportedTimestamp) {
 		// marker before minSupportedTimestamp
 		if !t.tangle.ConfirmationOracle.IsMessageConfirmed(messageID) {
 			// if unconfirmed, then incorrect
 			markerWalker.StopWalk()
-			return false
+			return messageID, false
 		}
-
-		// if closest past marker is confirmed and before minSupportedTimestamp, then message should be ok
-		return true
+		// if closest past marker is confirmed and before minSupportedTimestamp, then need to walk message past cone of the previously marker message
+		messageWalker.Push(previousMessageID)
```

Comment:
Needed!

Reply:
wdym?

Reply:
We needed to push the message to the messageWalker to evaluate it; before we were just assuming that if the closest marker was confirmed the message was ok.